src/main/java/com/gradle/develocity/bamboo/DevelocityPreBuildAction.java

```diff
@@ -38,25 +46,46 @@ public ErrorCollection validate(BuildConfiguration configuration) {
     @NotNull
     @Override
     public BuildContext call() {
-        for (BuildScanInjector<? extends BuildToolConfiguration> injector : injectors) {
-            try {
-                injector.inject(buildContext);
-            } catch (Exception e) {
-                LOGGER.error("Develocity {} auto-injection failed", injector.buildTool().displayName(), e);
+        if (injectionIsAllowedOnVcsRepo()) {
+            for (BuildScanInjector<? extends BuildToolConfiguration> injector : injectors) {
+                try {
+                    injector.inject(buildContext);
+                } catch (Exception e) {
+                    LOGGER.error("Develocity {} auto-injection failed", injector.buildTool().displayName(), e);
 
-                addErrorToBuildLog(injector, e);
+                    addErrorToBuildLog(injector, e);
+                }
             }
         }
 
         return buildContext;
     }
 
+    private boolean injectionIsAllowedOnVcsRepo() {
+        // TODO: we don't need GradleConfiguration
+        //  need to remove inheritance and use composition for BuildToolConfiguration, and instantiate that here
```

Comment:
Do you plan to resolve it in this PR?

Reply:
ah no, will add a comment, we'll refactor it if we encounter the same again